capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -107,12 +120,86 @@ public long upvotePost(HttpServletRequest request) {
 
     List<Entity> posts = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
 
-    int postID = Integer.parseInt(request.getParameter("id"));
-    Entity postToUpvote = posts.get(postID);
+    long postID = Long.parseLong(request.getParameter("id"));
```

Comment:
Can't you just retrieve the post by ID instead of fetching every post and scanning it for the right post?